packages/c/sshnpd/src/main.c

```diff
@@ -308,6 +327,14 @@ int main(int argc, char **argv) {
   main_loop();
   atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_INFO, "Exited main loop\n");
 
+  for(int i = 0; i < params.manager_list_len; i++) {
+    atclient_atkey_free(infokeys + i);
+    atclient_atkey_free(usernamekeys + i);
+  }
+
+  free(infokeys);
```

Comment:
Isn't some memory is lost here, because `infokeys + 0 == infokeys` and we overwrite the address of `infokeys + 0`, which means we lose the address to the original malloc in `infokeys`?

Reply:
I think that malloc is the `possibly lost: 272 bytes in 1 blocks`

Reply:
atclient_atkey_free only frees the memory that is initialized by atclient_atkey_init. (That's the pattern that is generally followed in the `init`/`free` pattern that we and MbedTLS follow). Any memory allocated by the caller should also be freed by them. For every malloc they malloc, they are responsible to free. For every init they call, they are responsible to free.

If `infokeys + 0` was statically allocated, then `atclient_atkey_free(infokeys + 0)` would cause an error

Reply:
Oh I see, atclient_atkey_init doesn't allocate new memory in place, doesn't free at the direct address passed in.